parameters/path_parameters.go

```diff
@@ -280,3 +281,27 @@ func (v *paramValidator) ValidatePathParams(request *http.Request) (bool, []*err
 	}
 	return true, nil
 }
+
+func (v *paramValidator) resolveNumber(sch *base.Schema, p *v3.Parameter, isLabel bool, isMatrix bool, paramValue string) (string, float64, []*errors.ValidationError) {
+	if isLabel && p.Style == helpers.LabelStyle {
+		paramValueParsed, err := strconv.ParseFloat(paramValue[1:], 64)
+		if err != nil {
+			return "", 0, []*errors.ValidationError{errors.IncorrectPathParamNumber(p, paramValue[1:], sch)}
+		}
+		return paramValue[1:], paramValueParsed, nil
+	}
+	if isMatrix && p.Style == helpers.MatrixStyle {
+		// strip off the colon and the parameter name
+		paramValue = strings.Replace(paramValue[1:], fmt.Sprintf("%s=", p.Name), "", 1)
+		paramValueParsed, err := strconv.ParseFloat(paramValue, 64)
+		if err != nil {
+			return "", 0, []*errors.ValidationError{errors.IncorrectPathParamNumber(p, paramValue[1:], sch)}
+		}
+		return paramValue, paramValueParsed, nil
+	}
+	paramValueParsed, err := strconv.ParseFloat(paramValue, 64)
```

Comment:
this is already parsed and checked there https://github.com/emilien-puget/libopenapi-validator/blob/c16b8e7fd4972fcf7c7beb2c612857f7cd910927/paths/paths.go#L291, should we validate during the path resolve for simple case ? (not object) @daveshanley 
the error could be path not found instead too

Reply:
I am wondering if path parameters validation must take place during path matching with the request or after, I'll take a look on how it is done in other system

Reply:
@daveshanley i'd suggest to remove the path param validation during the find path process and only validate the parameters when validating the request, wdyt ?

Reply:
OK, sounds good.